modules/misc/src/postprocessors/GeneralSensorPostprocessor.C

```diff
@@ -0,0 +1,317 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+#include <vector>
+#include <cmath>
+#include <algorithm>
+#include <numeric>
+
+#include "GeneralSensorPostprocessor.h"
+#include "Function.h"
+#include "MooseRandom.h"
+#include "LinearInterpolation.h"
+#include "SplineInterpolation.h"
+
+registerMooseObject("MooseApp", GeneralSensorPostprocessor);
+
+InputParameters
+GeneralSensorPostprocessor::validParams()   
+{
+  InputParameters params = GeneralPostprocessor::validParams();
+  params.addRequiredParam<PostprocessorName>("input_signal", "The input signal postprocessor");
+  params.addParam<FunctionName>("drift_function", 0.0,
+                                "Drift function describing signal drift over time");
+  params.addParam<Real>("vector_size", 1000000000.0, "The maximum size of vector to be saved");
+  params.addClassDescription("This is a GeneralSensorPostprocessor, and functions as a base class "
+                             "for other sensor postprocessors");
+  params.addParam<FunctionName>("efficiency_function", 0.8,
+                                  "Efficiency function describing efficiency over time");
+  params.addParam<FunctionName>("noise_std_dev_function", 0.5,
+                                "Standard deviation of noise function describing noise std dev over time");
+  params.addParam<FunctionName>("signalToNoise_function", 0.2,
+                                "Conversion of signal to noise function describing the conversion rate");

```

Comment:
```suggestion
  params.addParam<FunctionName>("signal_to_noise_ratio", 0.2,
                                "Signal to noise ratio for the modeled sensor");

```

I'd drop all the "_function" postfixes. The parameter type makes it clear that a function can be supplied.